lib/spack/spack/util/environment.py

```diff
@@ -745,11 +745,10 @@ def from_sourcing_file(
             ]
         )
 
+        no_op = "cd %CD%" if sys.platform == "win32" else os.devnull
```

Comment:
I think we need a comment here as well, an actual shell command on Windows or a null FD on Unix is a bit of disparate choice. 

(nit pick) Can we also rename the variable to something like `null_fd` or similar? That way in addition to the comment the variable name conveys what it's supposed to achieve.